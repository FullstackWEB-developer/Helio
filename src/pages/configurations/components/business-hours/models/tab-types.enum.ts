export enum TabTypes {
    GeneralSetting = 0,
    SetBusinessHours,
    SetHoliday,
}
