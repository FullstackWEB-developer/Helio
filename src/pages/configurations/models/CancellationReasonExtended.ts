import { CancellationReasonTypes } from "@pages/external-access/models/cancellation-reason-types.enum";

export interface CancellationReasonExtended {
    description?: string;
    existsOnEMR: boolean;
    id: string;
    intentName?: string;
    isDefault: boolean;
    isMapped: boolean;
    name: string;
    nameOnEMR?: string;
    type?: CancellationReasonTypes,
    createdByName: string,
    createdOn: Date,
    modifiedByName: string,
    modifiedOn?: Date
}