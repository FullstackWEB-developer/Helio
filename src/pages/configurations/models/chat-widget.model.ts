import { DisplayPosition } from "./display-position.enum";

export interface ChatWidgetModel
{
    displayPosition: DisplayPosition,
    domains: string[],
    autoStartEnabled: boolean,
    autoStartDelay: number
}