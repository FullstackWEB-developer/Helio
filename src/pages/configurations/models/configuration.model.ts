export interface ConfigurationModel {
    id: string;
    name: string;
    description: string;
    existsOnEMR: boolean;
    isMapped: boolean;
}
