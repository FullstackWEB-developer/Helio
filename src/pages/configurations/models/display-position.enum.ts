export enum DisplayPosition {
    RM = 1,
    RB
}
