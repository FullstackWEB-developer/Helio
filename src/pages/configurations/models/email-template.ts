export interface EmailTemplate {
    id: string;
    subject: string;
    templateBody: string;
    layoutName: string;
    name: string;
    description: string;
    title: string;
    defaultBody: string;
    body: string;
    createdByName: string;
    createdOn: Date;
    modifiedByName?: string;
    modifiedOn?: Date;
}