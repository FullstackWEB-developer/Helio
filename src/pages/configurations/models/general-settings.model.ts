export interface GeneralSettingsModel {
    forceToRedirect: boolean;
    redirectToExternalPhone: string;
    infoEmailAddress: string;
    infoPhoneNumber: string;
    deletedDepartments: string;
}