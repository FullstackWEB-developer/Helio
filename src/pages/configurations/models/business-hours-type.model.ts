import { DayOfWeek } from '@shared/models/DayOfWeek';

export enum WorkingHoursType {
  _24_7 = 1,
  Custom = 2,
}

export interface OperationSettingModel {
  id: number;
  timeZone: string;
  voiceRouting: boolean;
  autoReplySms: boolean;
  autoReplyEmail: boolean;
  routingNumber: string;
  workingHoursType: WorkingHoursType;
  workingHours: WorkingHourModel[];
  offDates: WorkingOffDateModel[];
}

export interface WorkingHourModel {
  day: DayOfWeek;
  startTime: string;
  endTime: string;
}

export interface WorkingOffDateModel {
  description: string;
  startDateTime: string;
  endDateTime: string;
}
