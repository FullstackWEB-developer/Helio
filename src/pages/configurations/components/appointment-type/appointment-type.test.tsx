import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import AppointmentType from './appointment-type';
import { AppointmentTypeSummary } from '@pages/appointments/models/appointment-type-summary';
import Api from '@shared/services/api';
import { act } from 'react-dom/test-utils';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
}));
describe("AppointmentType tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders AppointmentType correctly", async () => {
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <AppointmentType/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders AppointmentType with data correctly", async () => {
        var response = {
            data: [{
                id: "string1",
                name: "string",
                description: "string",
                existsOnEmr: false,
                isMapped: true,
                nameOnEmr: "string",
            },{
                id: "string2",
                name: "string",
                description: "string",
                existsOnEmr: false,
                isMapped: false,
                nameOnEmr: "string",
            },{
                id: "string3",
                name: "string",
                description: "string",
                existsOnEmr: true,
                isMapped: false,
                nameOnEmr: "string",
            },{
                id: "string4",
                name: "string",
                description: "string",
                existsOnEmr: true,
                isMapped: true,
                nameOnEmr: "string",
            }] as AppointmentTypeSummary[] 
        } as any
        jest.spyOn(Api, 'get').mockResolvedValue(response);
        await act(async () => {
            render(<TestWrapper mockState={mockState}>
                <AppointmentType/>
            </TestWrapper>);
        });
        expect(Api.get).toHaveBeenCalledTimes(1);
    });
})
