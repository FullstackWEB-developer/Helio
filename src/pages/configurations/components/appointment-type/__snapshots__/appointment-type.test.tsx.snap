// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentType tests renders AppointmentType correctly 1`] = `
<DocumentFragment>
  <div
    class="appointment-type overflow-auto h-full p-6 pr-4"
  >
    <h5
      class="appointment-type-header"
    >
      configuration.appointment_type_list.title
    </h5>
    <div
      class="body2"
    >
      configuration.appointment_type_list.description
    </div>
    <div
      class="flex flex-col space-y-4 items-center justify-center pt-2"
    >
      <div>
        <div
          class=""
        >
          <svg
            class="spinner icon-large-40 "
            viewBox="0 0 40 40"
          >
            <circle
              class="path"
              cx="20"
              cy="20"
              fill="none"
              r="16"
              stroke-width="4"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
